pandas/core/frame.py

```diff
@@ -2734,30 +2734,32 @@ def duplicated(self, subset=None, take_last=False):
         -------
         duplicated : Series
         """
-        # kludge for #1833
-        def _m8_to_i8(x):
-            if issubclass(x.dtype.type, np.datetime64):
-                return x.view(np.int64)
-            return x
+        from pandas.core.groupby import get_group_index
+        from pandas.hashtable import duplicated_int64, _SIZE_HINT_LIMIT
+
+        size_hint = min(len(self), _SIZE_HINT_LIMIT)
+
+        def factorize(vals):
```

Comment:
True, but I still find it clearer to use public APIs internally when possible (especially to avoid duplicated code).